pkg/cmd/pulumi/util.go

```diff
@@ -862,10 +865,43 @@ func getRefreshOption(proj *workspace.Project, refresh string) (bool, error) {
 		return true, nil
 	}
 
-	// the default functionality right now is to always skip a refresh
+	// if the user has passed a plan file then the default behaviour is to refresh
+	if planFilePath != "" {
+		return true, nil
+	}
+
+	// else the default functionality right now is to always skip a refresh
 	return false, nil
 }
 
+func writePlan(path string, plan deploy.Plan, enc config.Encrypter, showSecrets bool) error {
+	f, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer contract.IgnoreClose(f)
+
+	deploymentPlan, err := stack.SerializePlan(plan, enc, showSecrets)
+	if err != nil {
+		return err
+	}
+	return json.NewEncoder(f).Encode(deploymentPlan)
```

Comment:
This is not pretty-printing - so we get one long line.  We should likely do similar to https://github.com/pulumi/pulumi/blob/fraser/planPrototype/sdk/go/common/workspace/creds.go#L227:L227 to ensure we get pretty-printed JSON output here for readability.